### ISSUE TITLE
MPS: handshake layer restructuring

### ISSUE THREAD

**Issue by mpg** [Mbed-TLS/mbedtls#5768]
Handshake functions should be broken into steps, to separate writing/parsing from context update. See https://github.com/Mbed-TLS/mbedtls/pull/1634 which did that - except for practical reasons it's probably better when re-doing it to make it a series of smaller PRs.

**Comment by hanno-arm** [comment]
IIRC the TLS 1.3 code already splits the handshake functions, so it's only the 1.2 code that needs adjusting.

**Comment by hanno-arm** [comment]
Looking at #1634, I am not sure about the size estimate though -- I remember #1634 being rather tedious.

**Comment by mpg** [comment]
Yes, absolutely, I didn't mean S, my fingers just did that automatically.
